rules/coding-style/src/Rector/Property/UnderscoreToPascalCasePropertyNameRector.php

```diff
@@ -71,32 +71,39 @@ public function refactor(Node $node): ?Node
             return null;
         }
 
-        if (Strings::startsWith($nodeName, '_')) {
+        /** @var string $class */
+        $class = $node->getAttribute(AttributeKey::CLASS_NAME);
+        // properties are accessed via magic, nothing we can do
+        if (method_exists($class, '__set') || method_exists($class, '__get')) {
             return null;
         }
 
-        if (! Strings::contains($nodeName, '_')) {
+        // @see https://regex101.com/r/dXc89K/1
+        // matches if underscore is present but excluding first underscore
+        if (! Strings::match($nodeName, '#(?<!^)\_#')) {
             return null;
         }
 
         $camelCaseName = $this->createPascalName($nodeName, $node);
-        if ($camelCaseName === 'this') {
-            return null;
-        }
 
         $node->name = new Identifier($camelCaseName);
 
         return $node;
     }
 
     /**
-     * @param Variable|PropertyProperty|PropertyFetch|StaticPropertyFetch $node
+     * @param PropertyProperty|PropertyFetch|StaticPropertyFetch $node
      */
     private function createPascalName(string $nodeName, Node $node): string
```

Comment:
This method should be in `StaticRectorStrings` service with others

Reply:
1. I have just notice that it is possible to pass only the node as a param in this method and to resolve node name using `$this->getName($node)`. Is there a way to resolve the name of the node within `StaticRectorStrings`?

Reply:
2. I moved it but, in my opinion, there should be better place or name for it. There is already a method `underscoreToPascal`, also it is not clear from which naming convention it is going to change the name.
The current `StaticRectorStrings` seems like some helper class for strings. The new method is more related to the Node itself... at least I have such a feeling. Maybe it will be nice to create StaticRectorPropertyVariable and place this new method there with self explanotary name like `createPascalCaseNameFromUnderscore`...